src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs

```diff
@@ -391,6 +389,160 @@ void M()
     {
         int {|Rename:v|} = 1 + 1;
     }
+}");
+        }
+
+        [Fact, WorkItem(39537, "https://github.com/dotnet/roslyn/issues/39537")]
+        public async Task IntroduceDeconstruction1()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+
+class C
+{
+    (string someString, int someInt) X() => default;
+
+    void M()
+    {
+        X()[||]
+    }
+}",
+@"
+using System;
+
+class C
+{
+    (string someString, int someInt) X() => default;
+
+    void M()
+    {
+        var (someString, someInt) = X();
+    }
+}");
+        }
+
+        [Fact, WorkItem(39537, "https://github.com/dotnet/roslyn/issues/39537")]
+        public async Task IntroduceDeconstruction2()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+
+class C
+{
+    (string someString, int someInt) X() => default;
+
+    void M()
+    {
+        X();[||]
+    }
+}",
+@"
+using System;
+
+class C
+{
+    (string someString, int someInt) X() => default;
+
+    void M()
+    {
+        var (someString, someInt) = X();
+    }
+}");
+        }
+
+        [Fact, WorkItem(39537, "https://github.com/dotnet/roslyn/issues/39537")]
+        public async Task IntroduceDeconstruction3()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+
+class C
+{
+    (string someString, int someInt) X() => default;
+
+    void M()
+    {
+        X()[||]
+
+        string someString;
+    }
+}",
+@"
+using System;
+
+class C
+{
+    (string someString, int someInt) X() => default;
+
+    void M()
+    {
+        var (someString1, someInt) = X();
+
+        string someString;
+    }
+}");
+        }
+
+        [Fact, WorkItem(39537, "https://github.com/dotnet/roslyn/issues/39537")]
+        public async Task IntroduceDeconstruction4()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+
+class C
+{
+    ValueTuple<string, int> X() => default;
+
+    void M()
+    {
+        X()[||]
+    }
+}",
+@"
+using System;
+
+class C
+{
+    ValueTuple<string, int> X() => default;
+
+    void M()
+    {
+        (string, int) {|Rename:value|} = X();
+    }
+}");
+        }
+
+        [Fact, WorkItem(39537, "https://github.com/dotnet/roslyn/issues/39537")]
+        public async Task IntroduceDeconstruction5()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+
+class C
+{
+    (string, int) X() => default;
+
+    void M()
+    {
+        X()[||]
+    }
+}",
+@"
+using System;
+
+class C
+{
+    (string, int) X() => default;
+
+    void M()
+    {
+        (string, int) {|Rename:value|} = X();
```

Comment:
💭 Another option is `var (item1, item2)`

Reply:
i particularly didn't like that.  :)  i felt the `item1`/`item2` would be worse than today :)

Reply:
ok.  this is better now.  if it's going to be one of those names, we attempt to pick good local names based on type. e.g. `Customer customer`.